### PR TITLE
ACA-1842: VM Image SKUs support in Ansible collection for Azure module creation 

### PR DESCRIPTION
**Pull request by Yaish25491**
##### SUMMARY
Creating azure_rm_imagesku_info module to get all of the available SKU options for VM image. 
Creating azure_test_image_skus to test the creation of a VM with an image from the SKU list of options 
Updating the test list in the inventory file with azure_test_imageskus test.

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
azure_rm_imagesku_info.py
azure_test_image_skus.yml
azure_rm_virtualmachine/inventory.yml

### CODE REVIEW

File: plugins/modules/azure_rm_imagesku_info.py [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+        self.id = None
+        self.name = None
+        self.properties = None
+        self.tags = None
+
+        super(AzureRMImageskuInfo, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                               supports_check_mode=True,
+                                               supports_tags=False)
+
+    def list_skus(self):
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_imagesku_info.py:172]: To be consistent with other module templates, place exec_module before list_skus. Thank you!

**Comment by Fred-sun** [comment]
@Yaish25491 Please fix the below error! Thanks!
 ERROR: Found 8 pep8 issue(s) which need to be resolved:
ERROR: plugins/modules/azure_rm_imagesku_info.py:30:161: E501: line too long (177 > 160 characters)
ERROR: plugins/modules/azure_rm_imagesku_info.py:34:161: E501: line too long (199 > 160 characters)
ERROR: plugins/modules/azure_rm_imagesku_info.py:77:161: E501: line too long (192 > 160 characters)
ERROR: plugins/modules/azure_rm_imagesku_info.py:168:48: E128: continuation line under-indented for visual indent
ERROR: plugins/modules/azure_rm_imagesku_info.py:169:48: E128: continuation line under-indented for visual indent
ERROR: plugins/modules/azure_rm_imagesku_info.py:177:73: E128: continuation line under-indented for visual indent
ERROR: plugins/modules/azure_rm_imagesku_info.py:178:73: E128: continuation line under-indented for visual indent
ERROR: plugins/modules/azure_rm_imagesku_info.py:184:1: W293: blank line contains whitespace

File: plugins/modules/azure_rm_imagesku_info.py (hunk 1 above)
Comment (Yaish25491) [reply plugins/modules/azure_rm_imagesku_info.py:172]: Hi Fred-sun, I relocated exed_module to be above list_skus as requested

Comment (Fred-sun) [reply plugins/modules/azure_rm_imagesku_info.py:172]: But now I haven't seen the latest commit, thank you!

**Comment by Yaish25491** [comment]
Hi Fred-Sun,
All errors corrected and pushed again for review

**Comment by Fred-sun** [comment]
@Yaish25491 

ERROR: plugins/modules/azure_rm_imagesku_info.py:0:0: invalid-ansiblemodule-schema: AnsibleModule.argument_spec.properties.automaticOSUpgradeProperties: extra keys not allowed @ data['argument_spec']['properties']['automaticOSUpgradeProperties']. Got {'type': 'bool', 'default': 'false'}
ERROR: plugins/modules/azure_rm_imagesku_info.py:0:0: nonexistent-parameter-documented: Argument 'publisher' is listed in DOCUMENTATION.options, but not accepted by the module argument_spec
ERROR: plugins/modules/azure_rm_imagesku_info.py:0:0: parameter-type-not-in-doc: Argument 'id' in argument_spec defines type as 'str' but documentation doesn't define type
ERROR: plugins/modules/azure_rm_imagesku_info.py:0:0: parameter-type-not-in-doc: Argument 'name' in argument_spec defines type as 'str' but documentation doesn't define type
ERROR: plugins/modules/azure_rm_imagesku_info.py:0:0: parameter-type-not-in-doc: Argument 'properties' in argument_spec defines type as 'str' but documentation doesn't define type
ERROR: plugins/modules/azure_rm_imagesku_info.py:0:0: parameter-type-not-in-doc: Argument 'publisher_name' in argument_spec defines type as 'str' but documentation doesn't define type
ERROR: plugins/modules/azure_rm_imagesku_info.py:0:0: parameter-type-not-in-doc: Argument 'tags' in argument_spec defines type as 'str' but documentation doesn't define type
ERROR: plugins/modules/azure_rm_imagesku_info.py:0:0: undocumented-parameter: Argument 'id' is listed in the argument_spec, but not documented in the module documentation
ERROR: plugins/modules/azure_rm_imagesku_info.py:0:0: undocumented-parameter: Argument 'name' is listed in the argument_spec, but not documented in the module documentation
ERROR: plugins/modules/azure_rm_imagesku_info.py:0:0: undocumented-parameter: Argument 'properties' is listed in the argument_spec, but not documented in the module documentation
ERROR: plugins/modules/azure_rm_imagesku_info.py:0:0: undocumented-parameter: Argument 'publisher_name' is listed in the argument_spec, but not documented in the module documentation
ERROR: plugins/modules/azure_rm_imagesku_info.py:0:0: undocumented-parameter: Argument 'tags' is listed in the argument_spec, but not documented in the module documentation
ERROR: plugins/modules/azure_rm_imagesku_info.py:30:99: W291: trailing whitespace
ERROR: plugins/modules/azure_rm_imagesku_info.py:35:133: W291: trailing whitespace
ERROR: plugins/modules/azure_rm_imagesku_info.py:200:1: E302: expected 2 blank lines, found 1

**Comment by Yaish25491** [comment]
issues fixed and merged

File: tests/integration/targets/azure_rm_virtualmachine/tasks/azure_test_image_skus.yml [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    name: "{{ subnet_name }}"
+    address_prefix: "{{ subnet }}"
+    virtual_network: "{{ network_name }}"
+
+- name: Create VM with first Image SKU result
+  azure_rm_virtualmachine:
+    resource_group: "{{ resource_group }}"
+    name: "{{ vm_name }}"
+    admin_username: "testuser"
+    ssh_password_enabled: false
```
Comment (Fred-sun) [inline tests/integration/targets/azure_rm_virtualmachine/tasks/azure_test_image_skus.yml:49]: Please set the open ports, Now , The default 22 port was disallowed by policy！
```suggestion
    ssh_password_enabled: false
    open_ports:
      - 33
```

Comment (Yaish25491) [reply tests/integration/targets/azure_rm_virtualmachine/tasks/azure_test_image_skus.yml:49]: Hi, Fred-sun.
Change applied, Please confirm and merge.

**Review by xuzhang3 — APPROVED** [review]

**Pull request merged**